text/0000-execution-environments.md

```diff
@@ -0,0 +1,225 @@
+# Meta
+[meta]: #meta
+- Name: Execution Environments
+- Start Date: 2023-01-11
+- Author(s): @hone
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+Add support for different execution environments for buildpacks, with testing as the first use case.
+
+# Definitions
+[definitions]: #definitions
+
+- Execution Environment - The target environment the OCI image is expected to be run in, i.e. production, test, development.
+
+# Motivation
+[motivation]: #motivation
+
+The main way Buildpacks are being used is building production images, but this is only one piece of the CI/CD pipeline. Without a solid buildpack test environment story, users will be required alternatives for building their environment. A testing environment, while different, shares many of the same broad strokes ultimately producing an execution environment. This is something Buildpacks are well suited to solve.
```

Comment:
```suggestion
The main way Buildpacks are being used is building production images, but this is only one piece of the software development process. Without a solid buildpack test environment story, users will be required alternatives for building their environment. A testing environment, while different, shares many of the same broad strokes ultimately producing an execution environment. This is something Buildpacks are well suited to solve.
```

I'm thinking this RFC could be useful outside of CI/CD pipelines, most notably for testing changes locally. However, I could also imagine building development environments locally with features like hot reload, auto recompilation, web servers that respond with stack traces, etc.